### PR TITLE
[FIX] Monetary widget throws error without aggregation in list view

### PR DESCRIPTION
**Pull request by EluciferE**
### Impacted versions:
16.0

### Steps to reproduce:
1. Create list view
2. Use the monetary widget for a column with various currencies.
3. Do not aggregate the column.

### Bug Fix Description:

**Bug Description:**

Previously, in the `web` module's `list_render`, an issue occurred where an error message was incorrectly being displayed below a table column with multiple currencies, even when the column wasn't being aggregated. This behavior was misleading and could confuse users, as the error message appeared even when it wasn't relevant.

**Root Cause:**

The error message was triggered regardless of whether the column was being aggregated by the user. This resulted in incorrect behavior and an unnecessary error display.

**Resolution:**

In this PR, the issue has been addressed by modifying the error display logic. The error message will no longer appear if the user does not choose to aggregate the column. This enhancement ensures that the error message is contextually relevant and only displayed when aggregation is being attempted on the column.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/tests/views/list_view_tests.js [hunk 1]
```diff
@@ -4012,8 +4012,9 @@ QUnit.module("Views", (hooks) => {
             );
             assert.strictEqual(
                 target.querySelectorAll("tfoot td")[1].textContent,
-                "—",
-                "aggregates monetary should never work if different currencies are used"
+                "",
+                "monetary aggregation should only be attempted with an active aggregation function" +
+                " when using different currencies"
```
Comment (juliusc2066) [inline addons/web/static/tests/views/list_view_tests.js:4017]: I would suggest to keep this part but add the ` sum="Sum"` property to the "amount field line 3998 instead and add another assertion for your fix

File: addons/web/static/tests/views/list_view_tests.js [hunk 2]
```diff
@@ -4012,8 +4012,9 @@ QUnit.module("Views", (hooks) => {
...
             );
```
Comment (juliusc2066) [inline addons/web/static/tests/views/list_view_tests.js:4018]:
```suggestion
            );
            assert.strictEqual(
                target.querySelectorAll("tfoot td")[2].textContent,
                "",
                "monetary aggregation should only be attempted with an active aggregation function" +
                " when using different currencies"
            );
```
should do the work to test your fix

**Comment by juliusc2066** [comment]
Good job overall :)

**Comment by EluciferE** [comment]
Thanks for the answer. I took into account your comments and updated the code

**Comment by juliusc2066** [comment]
Hello @EluciferE, sorry for the late response but I didn't notice this being forgotten :bow:. To make things simpler for you, your changes are now handled in https://github.com/odoo/odoo/pull/183730 so that you won't have to worry about rebase and forward ports. Have a nice day :smile:

**Pull request closed without merging**